app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/Adapters/SectionAdapter.java

```diff
@@ -0,0 +1,124 @@
+        mMapsPresenter = clickHandler;
+        mCount = list.length;
+        mRoutesList = list;
+        mTitle = title;
+        mType = type;
+    }
+
+    public void setList(Route[] list, int count, String query) {
+        mCount = list.length;
+        mRoutesList = list;
+    }
+
+    @Override
+    public int getContentItemsTotal() {
+        return mCount;
+    }
+
+    @Override
+    public RecyclerView.ViewHolder getItemViewHolder(View view) {
+        return new ItemHolder(view);
+    }
+
+    @Override
+    public void onBindItemViewHolder(RecyclerView.ViewHolder holder,
+            int position) {
+        Route routeModel = mRoutesList[position];
+
+        ItemHolder holder2 = (ItemHolder) holder;
+        holder2.duration.setText(routeModel.getDuration());
+        holder2.route_description.setText(routeModel.getDescription());
+        holder2.rootView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                mMapsPresenter.onRouteClick(position, mRoutesList);
+            }
+        });
+        holder2.directions.setHasFixedSize(true);
+        LinearLayoutManager layoutManager = new LinearLayoutManager(mContext,
+                LinearLayoutManager.VERTICAL, false);
+        holder2.directions.setLayoutManager((RecyclerView.LayoutManager) layoutManager);
+        holder2.directions.setAdapter(
+                new DirectionsListAdapter(mContext, routeModel.getDirections()));
+
+    }
+
+    public interface ListAdapterOnClickHandler {
+        void onRouteClick(int position, Route[] routesList);
+    }
+
+    private class ItemHolder extends RecyclerView.ViewHolder {
+        private TextView duration;
+        private RecyclerView directions;
+        private View rootView;
+        private TextView route_description;
+
+
+        ItemHolder(View itemView) {
+            super(itemView);
+            duration = itemView.findViewById(R.id.duration);
+            directions = itemView.findViewById(R.id.directions);
+            rootView = itemView;
+            route_description = itemView.findViewById(R.id.route_description);
+        }
+
+    }
+
+    private class HeaderHolder extends RecyclerView.ViewHolder {
+        private TextView header;
+        private View rootView;
+
+
+        HeaderHolder(View headerView) {
+            super(headerView);
+            rootView = headerView;
+            header = headerView.findViewById(R.id.header_holder);
+        }
+    }
+
+    @Override
+    public void onBindHeaderViewHolder(RecyclerView.ViewHolder holder) {
+        HeaderHolder headerHolder = (HeaderHolder) holder;
+        headerHolder.header.setText(mTitle);
+
+        //TODO: If section is the second section, it's title should be "See all route options"
+        // until slide up
+    }
+
+    @Override
+    public RecyclerView.ViewHolder getHeaderViewHolder(View view) {
+        return new HeaderHolder(view);
+    }
+
+
```

Comment:
Bit of extra spacing here, and potentially edit the TODO on line 115 above to make it either one line or a multiline comment.